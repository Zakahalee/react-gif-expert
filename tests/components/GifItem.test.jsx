const { render, screen } = require("@testing-library/react");
const { GifItem } = require("../../src/components/GifItem");

describe('Pruebas en <GifItem/>', () => { 
    const title = 'Saitama';
    const url = 'https://google.com.mx';

    test('should match with snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })

     test('should show image and indicated URL', () => { 
        render(<GifItem title={title} url={url}/>);
        //screen.debug();
        //expect(screen.getByRole('img').src).toContain(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toContain(url);
        expect(alt).toBe(title);
      });

    test('should show title in the component', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
     })
 })