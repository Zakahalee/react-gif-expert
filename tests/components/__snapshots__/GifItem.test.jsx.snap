// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://google.com.mx"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
